src/plugins/usage_collection/server/routes/stats.ts

```diff
@@ -0,0 +1,191 @@
+          ),
+          exclude_usage: schema.maybe(
+            schema.oneOf([schema.literal(''), schema.literal('true'), schema.literal('false')])
+          ),
+        }),
+      },
+    },
+    async (context, req, res) => {
+      const isExtended = req.query.extended === '' || req.query.extended === 'true';
+      const isLegacy = req.query.legacy === '' || req.query.legacy === 'true';
+      const shouldGetUsage =
+        req.query.exclude_usage === undefined || req.query.exclude_usage === 'false';
+
+      let extended;
+      if (isExtended) {
+        const callCluster = context.core.elasticsearch.legacy.client.callAsCurrentUser;
+        const collectorsReady = await collectorSet.areAllCollectorsReady();
+
+        if (shouldGetUsage && !collectorsReady) {
+          return res.customError({ statusCode: 503, body: { message: STATS_NOT_READY_MESSAGE } });
+        }
+
+        const usagePromise = shouldGetUsage ? getUsage(callCluster) : Promise.resolve({});
+        const [usage, clusterUuid] = await Promise.all([usagePromise, getClusterUuid(callCluster)]);
+
+        let modifiedUsage = usage;
+        if (isLegacy) {
+          // In an effort to make telemetry more easily augmented, we need to ensure
+          // we can passthrough the data without every part of the process needing
+          // to know about the change; however, to support legacy use cases where this
+          // wasn't true, we need to be backwards compatible with how the legacy data
+          // looked and support those use cases here.
+          modifiedUsage = Object.keys(usage).reduce((accum, usageKey) => {
+            if (usageKey === 'kibana') {
+              accum = {
+                ...accum,
+                ...usage[usageKey],
+              };
+            } else if (usageKey === 'reporting') {
+              accum = {
+                ...accum,
+                xpack: {
+                  ...accum.xpack,
+                  reporting: usage[usageKey],
+                },
+              };
+            } else {
+              // I don't think we need to it this for the above conditions, but do it for most as it will
+              // match the behavior done in monitoring/bulk_uploader
+              defaultsDeep(accum, { [usageKey]: usage[usageKey] });
+            }
+
+            return accum;
+          }, {} as any);
+
+          extended = {
+            usage: modifiedUsage,
+            clusterUuid,
+          };
+        } else {
+          extended = collectorSet.toApiFieldNames({
+            usage: modifiedUsage,
+            clusterUuid,
+          });
+        }
+      }
+
+      // Guranteed to resolve immediately due to replay effect on getOpsMetrics$
+      const lastMetrics = await metrics.getOpsMetrics$().pipe(first()).toPromise();
+
+      const overallStatus = await overallStatus$.pipe(first()).toPromise();
+      const kibanaStats = collectorSet.toApiFieldNames({
+        ...lastMetrics,
```

Comment:
Testing it locally, I've noticed the response contains `"collected_at": {},`. We might want to deconstruct it from `lastMetrics`. What do you think?